python/tank/platform/software_launcher.py

```diff
@@ -85,16 +89,28 @@ class SoftwareLauncher(object):
     should only be constructed through the :meth:`create_engine_launcher()`
     factory method.
     """
-    def __init__(self, tk, context, engine_name, env):
+    def __init__(self, tk, context, engine_name, env, versions=None, products=None):
         """
         :param tk: :class:`~sgtk.Sgtk` Toolkit instance
-        :param context: :class:`~sgtk.Context` A context object to
-                        define the context on disk where the engine
-                        is operating
+
+        :param context: :class:`~sgtk.Context` A context object to define the
+            context on disk where the engine is operating
+
         :param str engine_name: Name of the Toolkit engine associated
-                                with the DCC(s) to launch.
+            with the DCC(s) to launch.
+
         :param env: An :class:`~sgtk.platform.environment.Environment` object to
-                    associate with this launcher.
+            associate with this launcher.
+
+        :param list versions: List of strings representing versions to search
+            for. If set to None or [], search for all versions. A version string
```

Comment:
We tend to double backtick stuff like `None` or `[]` in the doc so these keywords stand out more.